src/monitoring/aws/lambda/metrics.ts

```diff
@@ -12,59 +12,33 @@ const Namespace = 'AWS/Lambda';
 
 export class LambdaMetricFactory {
   metricInvocations(functionName: string) {
-    return new Metric({
-      metricName: Metrics.Invocations,
-      namespace: Namespace,
-      period: cdk.Duration.minutes(5),
-      statistic: Statistic.SUM,
-      dimensions: {
-        FunctionName: functionName,
-      },
-    });
+    return this.metric(Metrics.Invocations, functionName).with({ statistic: Statistic.SUM });
   }
 
   metricDuration(functionName: string) {
-    function metric(statistic: string) {
-      return new Metric({
-        metricName: Metrics.Duration,
-        label: statistic,
-        namespace: Namespace,
-        period: cdk.Duration.minutes(5),
-        statistic,
-        dimensions: {
-          FunctionName: functionName,
-        },
-      });
-    }
-
     return {
-      min: metric(Statistic.MINIMUM),
-      avg: metric(Statistic.AVERAGE),
-      p50: metric('p50'),
-      p90: metric('p90'),
-      p99: metric('p99'),
-      max: metric(Statistic.MAXIMUM),
+      min: this.metric(Metrics.Duration, functionName).with({ statistic: Statistic.MINIMUM, label: Statistic.MINIMUM }),
+      avg: this.metric(Metrics.Duration, functionName).with({ statistic: Statistic.AVERAGE, label: Statistic.AVERAGE }),
+      p50: this.metric(Metrics.Duration, functionName).with({ statistic: 'p50', label: 'p50' }),
+      p90: this.metric(Metrics.Duration, functionName).with({ statistic: 'p90', label: 'p90' }),
+      p99: this.metric(Metrics.Duration, functionName).with({ statistic: 'p99', label: 'p99' }),
+      max: this.metric(Metrics.Duration, functionName).with({ statistic: Statistic.MAXIMUM, label: Statistic.MAXIMUM }),
```

Comment:
How about we create a metric instance with common properties and extend them for others? Something like:

```ts
const durationMetric = this.metric(Metrics.Duration, functionName);

return {
	min: durationMetric.with({...}),
	...
};
```

Reply:
Hmmm, sounds good!